mlflow/__init__.py

```diff
@@ -66,15 +66,18 @@
 
 _configure_mlflow_loggers(root_module_name=__name__)
 
-if sys.version_info.major == 2:
+_version_to_deprecate = (3, 5)
+if sys.version_info[:2] == _version_to_deprecate:
     warnings.warn(
-        "MLflow support for Python 2 is deprecated and will be dropped in a future "
-        "release. At that point, existing Python 2 workflows that use MLflow will "
-        "continue to work without modification, but Python 2 users will no longer "
-        "get access to the latest MLflow features and bugfixes. We recommend that "
-        "you upgrade to Python 3 - see https://docs.python.org/3/howto/pyporting.html "
-        "for a migration guide.",
-        DeprecationWarning,
+        "MLflow support for Python {ver} is deprecated and will be dropped in "
+        "an upcoming release. At that point, existing Python {ver} workflows "
+        "that use MLflow will continue to work without modification, but Python {ver} "
+        "users will no longer get access to the latest MLflow features and bugfixes. "
+        "We recommend that you upgrade to Python 3.6 or newer.".format(
```

Comment:
Small nit: should we also express "3.6" here based on `_version_to_deprecate`? e.g. something like 

```
major, minor = _version_to_deprecate
min_supported_version = ".".join(map(str, [major, minor + 1]))
"We recommend you upgrade to Python {min_supported_version}".format(min_supported_version=min_supported_version)